### PR TITLE
Simplify petemaild

### LINKED ISSUE #42: Email sending stalls when domain resolving stalls. Should we simplify the email sending completely?
From @dracos:

> It seems it's possible for the `petemaild` daemon to still get stuck sometimes and stop
> trying to resolve/deliver messages, sigh.
> 
> From looking at the DB they were almost all aol.com email addresses, so
> it looked like it had got stuck on delivering to that domain.
> 
> From a 'kill -USR2' on the process, which writes a file to
> /tmp/petemaild.context.PID it looked like the DomainResolver had nothing
> on its queue, but one thing in its 'state' (which is only meant to be
> there for actively-resolving DomainResolver::Tasks). When the queue
> loops through pending messages, if a domain isn't resolved, it adds it
> to the DomainResolver queue:
> https://github.com/mysociety/petitions/blob/master/bin/petemaild#L1186-L1192
> which calls this function:
> https://github.com/mysociety/petitions/blob/master/bin/petemaild#L2422-L2433
> 
> However, that function won't add something to the queue if there's
> already a currently-processing 'state' for that domain (which makes
> sense if nothing goes wrong). So I guess the domain name lookup of
> 'aol.com' got 'stuck' somehow, neither succeeding nor failing, and so it
> never tried to look it up again as it always thought it was already
> doing so. It would seem like the Resolver needs a timeout then to deal
> with that; not entirely sure the simplest way of doing that, but I guess
> in DomainResolver's post_select you'd want it to know how long it has
> been trying and give up after some length of time. Or it looks like
> Net::DNS::Resolver has timeout functions, but I don't know what happens
> when they're met. That's if I'm guessing what the problem was correctly.
> 
> Frankly, given traffic levels, I wonder if it would be quicker just to
> replace most/all of petemaild with something that just loops through the
> existing queue, sends the emails to local exim and nothing else... (then
> replace petsignupd with rabbitmq or whatever, it feels like there's so
> much unnecessary code, hohum).

In the interim, `kill -USR2`ing petemaild and restarting it seems to have solved the queue filling up problem at the moment...

### PR DESCRIPTION
**Pull request by dracos**
Fixes #42. Does two things - removes the domain resolver, assuming there's a localhost email server. And then removes the email sender, replacing it with Email::Sender. Could do either/or of those if there were some reason we didn't want to do both. The test suite passed on firefly.

### CODE REVIEW

File: bin/petemaild [hunk 1]
```diff
@@ -1059,100 +330,25 @@ sub requeue ($$) {
...
     my IO::Select $rds = shift;
     my IO::Select $wrs = shift;
 
-    # Do pre-select handling for domain resolver and individual SMTP clients.
-    $self->r()->pre_select($rds, $wrs);
-    foreach my $smtp (@{$self->{ss}}) {
-        $smtp->pre_select($rds, $wrs);
-    }
-
     # Hash of IP addresses to which to send messages to messages to send to
```
Comment (sagepe) [inline bin/petemaild:340]: Remove this comment since the code to which it refers has gone?

Comment (dracos) [reply bin/petemaild:340]: Weird, how did I miss that.

**Comment by dracos** [comment]
"It looks fine - although this is a pretty large program, and I'm not hugely familiar with all its parts." - yeah, that's one good reason to do this, I think, as I find it pretty hard to follow too!
